cloudinit/net/eni.py

```diff
@@ -481,16 +493,60 @@ def _render_iface(self, iface, render_hwaddress=False):
                 ]:
                     iface["control"] = "alias"
 
+                # v1 config has the dns info in the first non-dhcp route,
+                # replicate dns to others to be more correct
+                dns_present = (
+                    "dns_search" in subnet or "dns_nameservers" in subnet
+                )
+                if dns is None and dns_present:
+                    dns = dict(
+                        (k, subnet.get(k))
+                        for k in ("dns_search", "dns_nameservers")
+                    )
+                if dns is not None and not dns_present:
+                    subnet = {**subnet, **dns}
+
                 lines = list(
                     _iface_start_entry(
                         iface, index, render_hwaddress=render_hwaddress
                     )
-                    + _iface_add_subnet(iface, subnet)
+                    + _iface_add_subnet(iface, subnet, is_ipv6)
                     + _iface_add_attrs(iface, index, ipv4_subnet_mtu)
                 )
                 for route in subnet.get("routes", []):
+                    ipv6_network = ":" in route.get("network", "")
```

Comment:
there's helpers in `cloudinit/net/__init__.py` like `is_ipv6_address`, `is_ipv6_network`, and `subnet_is_ipv6`. We should lean on these rather than to do it separately with checks like this.